### PR TITLE
Add support for azure workload identity authentication

### PR DESCRIPTION
**Pull request by cldmstr**
Use the Azure Workload Identity federated identity when it is available to authenticate against Azure keyvault. This allows vault to run in an AKS cluster with service accounts federated to service principals or managed identities.

The PR uses the default credential reader, which covers all the Azure login credentials: MSI, Azure CLI or Federated Identity.

This change also updates the Azure client libraries to use the azure-sdk-for-go implementation as the autorest libraries are being depricated at the end of March 2023.

### CODE REVIEW

File: wrappers/azurekeyvault/azurekeyvault.go [hunk 1]
```diff
@@ -166,14 +166,14 @@ func (v *Wrapper) SetConfig(_ context.Context, opt ...wrapping.Option) (*wrappin
 
 		if !v.keyNotRequired {
 			// Test the client connection using provided key ID
-			keyInfo, err := client.GetKey(context.Background(), v.baseURL, v.keyName, "")
+			keyInfo, err := client.GetKey(context.Background(), v.keyName, "", nil)
```
[the remark this answers is not in the document]
Comment (cldmstr) [reply wrappers/azurekeyvault/azurekeyvault.go]: Added the `ctx` pass-through

**Comment by AntPAllen** [comment]
Happy to test this if it helps. Came to this repo looking to see if it was possible to auth boundary using workload identity.

**Comment by jimlambrt** [comment]
Ty for the reminder.  Now that Boundary v0.13.0, we should be able to find a resource to test this.  I'll be back with an update on when that might happen.

**Comment by ibauersachs** [comment]
@jimlambrt Do you have an update on that resource or timeline? Is there anything we can help with?

**Comment by jimlambrt** [comment]
Good afternoon!  Yes, we've started to make some progress allocating a resource and building some sort of automated tests.  Updates in the next week or so 🤞

File: wrappers/azurekeyvault/azurekeyvault.go [hunk 2]
```diff
@@ -295,39 +280,35 @@ func (v *Wrapper) buildBaseURL() string {
...
-	// By default use MSI
+	// By default let Azure select existing credentials
 	default:
-		config := auth.NewMSIConfig()
-		config.Resource = strings.TrimSuffix(v.resource, "/")
-		if v.clientID != "" {
-			config.ClientID = v.clientID
-		}
-		authorizer, err = config.Authorizer()
+		cred, err = azidentity.NewDefaultAzureCredential(nil)
```
[the remark this answers is not in the document]
Comment (cldmstr) [reply wrappers/azurekeyvault/azurekeyvault.go:296]: Thanks for the review @jtv8 . I can confirm that the behavior in `azure-sdk-for-go` is the same as the fix and does not introduce a regression. Internally, the call does almost exactly what the fix does.

**Comment by jimlambrt** [comment]
We've been able to successfully validate/test this change!  @cldmstr can you resolve the conflicts so we can get it merged?

**Review by jimlambrt — APPROVED** [review]
ty for your patience.

**Comment by cldmstr** [comment]
Thanks a lot. There was one relevant change to the method I was working on that cropped up during the rebase. I did my best to add the fix to 
[Add Work around an azure HTTP/2 bug](https://github.com/hashicorp/go-kms-wrapping/pull/155/commits/07c69978a1d00f949e27e6275dad624ee647d5a0) in the new client. However, I couldn't test it quite yet myself.

**Pull request merged**